core/new-gui/src/app/workspace/component/property-editor/property-editor.component.ts

```diff
@@ -65,18 +74,27 @@ export class PropertyEditorComponent implements OnInit {
           .getCurrentHighlightedOperatorIDs();
         const highlightedGroups = this.workflowActionService.getJointGraphWrapper().getCurrentHighlightedGroupIDs();
         const highlightLinks = this.workflowActionService.getJointGraphWrapper().getCurrentHighlightedLinkIDs();
+        const workflowVersions = this.workflowVersionService.getWorkflowVersions();
 
-        if (highlightedOperators.length === 1 && highlightedGroups.length === 0 && highlightLinks.length === 0) {
+        if (workflowVersions.length > 0 && highlightLinks.length === 0 && highlightedGroups.length === 0 && highlightedOperators.length === 0) {
+          this.switchFrameComponent({
+            component: VersionsListDisplayComponent,
+            componentInputs: {currentLinkId: highlightLinks[0]},
+          });
+        } else if (highlightedOperators.length === 1 && highlightedGroups.length === 0 && highlightLinks.length === 0) {
+          this.workflowVersionService.resetResults();
```

Comment:
Don't think it's a good design to manually clear the results here. Here are some good principles:

If you want to fetch the versions only when the version list component is shown, and discard the versions when it's not shown, then store the version list inside the *component* not the service. So it's bind to the component lifecycle.

If you want the version list to save in the frontend even if the version list is not shown, then put the version list in the service.